lisp/mastodon-tl.el

```diff
@@ -513,13 +514,33 @@ LINK-TYPE is the type of link to produce."
      'keymap mastodon-tl--link-keymap
      'help-echo help-text)))
 
+(defun mastodon-tl--handle-to-account (handle)
+  "Return an account based on a users HANDLE.
+
+If the handle does not match a search return then retun NIL."
+  (let* ((handle (if (string= "@" (substring handle 0 1))
+                     (substring handle 1 (length handle))
+                   handle))
+         (matching-account
+          (remove-if-not
+           (lambda(x) (string= (cdr (assoc 'acct x)) handle))
+           (mastodon-http--get-json
+            (mastodon-http--api (format "accounts/search?q=%s" handle))))))
+    (when (equal 1 (length matching-account))
+      (elt matching-account 0))))
+
 (defun mastodon-tl--do-link-action-at-point (position)
   (interactive "d")
   (let ((link-type (get-text-property position 'mastodon-tab-stop)))
     (cond ((eq link-type 'content-warning)
            (mastodon-tl--toggle-spoiler-text position))
           ((eq link-type 'hashtag)
            (mastodon-tl--show-tag-timeline (get-text-property position 'mastodon-tag)))
+          ((eq link-type 'handle)
+           ;; replace message with mastodon-profile--make-author-buffer
+           (message "its a handle: %s"
```

Comment:
In my cursory testing this seems to work.
Get rid of the `message` and instead actually show the profile.